nautobot/core/views/mixins.py

```diff
@@ -1387,3 +1388,106 @@ def notes(self, request, *args, **kwargs):
             "active_tab": "notes",
         }
         return Response(data)
+
+
+class ComponentCreateViewMixin(NautobotViewSetMixin, mixins.CreateModelMixin):
+    create_form_class: Optional[type[Form]] = None  # TODO: required, declared Optional only to avoid a breaking change
+    form_class: Optional[type[Form]] = None  # TODO: required, declared Optional only to avoid a breaking change
+
+    def create(self, request, *args, **kwargs):
+        if request.method == "POST":
+            return self.perform_create(request, *args, **kwargs)
+
+        create_form = self.create_form_class(initial=request.GET)  # pylint: disable=not-callable
+        model_form = self.form_class(request.GET)  # pylint: disable=not-callable
```

Comment:
Addressing my comment above should also remove the need for the pylint disables here.